oci/oci.go

```diff
@@ -514,20 +514,34 @@ func (r *Runtime) StopContainer(c *Container, timeout int64) error {
 		timeout = 10
 	}
 	done := make(chan struct{})
+	// we could potentially re-use "done" channel to exit the loop on timeout
+	// but we use another channel "chControl" so that we won't never incur in the
+	// case the "done" channel is closed in the "default" select case and we also
+	// reach the timeout in the select below. If that happens we could raise
+	// a panic closing a closed channel so better be safe and use another new
+	// channel just to control the loop.
+	chControl := make(chan struct{})
 	go func() {
 		for {
-			// Check if the process is still around
-			err := unix.Kill(c.state.Pid, 0)
-			if err == syscall.ESRCH {
-				close(done)
-				break
+			select {
+			case <-chControl:
+				return
+			default:
+				// Check if the process is still around
+				err := unix.Kill(c.state.Pid, 0)
+				if err == syscall.ESRCH {
+					close(done)
+					return
+				}
+				time.Sleep(100 * time.Millisecond)
 			}
```

Comment:
If the process never goes away could we get stuck in this loop for ever?  i.e. if err is not syscall.ESRCH?  Should we do an else here, and do a fmt.Error like line 547 and then return?  Or perhaps loop some number of times and then do that?

I know it's probably overkill, but Murphy hates me.